content/blog/2021-01-26-january-21-community-gems.md

```diff
@@ -0,0 +1,184 @@
+tags:
+  - Discord
+  - Gems
+  - CML
+  - Plots
+  - Pipelines
+  - Docker
+---
+
+## DVC questions
+
+### [Q: My dataset is made of more than _a million_ small files. Can I use an archive format, like `tar.gz` with DVC?](https://discord.com/channels/485586884165107732/485596304961962003/798983422965841920)
+
+There are some downsides to using archive formats with DVC, and often we
+discourage it- but let's review some factors to consider, so you can make the
+best choice for your project.
+
+- DVC versions via snapshots, not diffs. So if your `tar.gz` file changes at
+  all- perhaps because you changed a single file before zipping- DVC will create
+  an entirely new copy of the archive for versioning! This is not very space
+  efficient, but if space isn't an issue it might not be a dealbreaker.
+- Because of the way DVC optimizes data transfer, you'll end up transferring the
+  whole archive anytime you modify a single file and `dvc push`/`dvc pull`.
+- If you're worried about data transfer speeds with the raw dataset, we
+  recommend exploring our parallelization options. DVC has parallelism built-in
+  and data transfer functions like `dvc push` and `dvc pull` have a flag (`-j`)
+  for increasing the number of jobs run simultaneously.
+  [Check out the docs for more details](https://dvc.org/doc/command-reference/push#options).
+
+In summary, the advantage of using an archive format will depend on both how
+often you modify your dataset and how often you need to push and pull data. You
+might consider exploring both approaches (with and without compression) and run
+some speed tests for your use case. We'd love to know what you find!
+
+### [Q: My DVC remote is a server with a self-signed certificate. When I push data, DVC is giving me an SSL verification error- how can I get around this?](https://discord.com/channels/485586884165107732/563406153334128681/800707271502856222)
+
+If you're using an `http`-addressable storage, try modifying your DVC remote
+configuration to disable SSH verification:
+
+```dvc
+$ dvc remote modify <myremote> ssl_verify false
+```
+
+If your storage is another format, you'll want to check into the manufacturer's
+docs to see how you can make your certificate pass verification. For example, if
+you use S3 or S3-compatible storage, you can configure your AWS CLI to use a
+custom certificate path.
+[As suggested by their docs](https://docs.aws.amazon.com/credref/latest/refdocs/setting-global-ca_bundle.html),
+you'll set the environmental variable `AWS_CA_BUNDLE` to your `.pem` file.
+
+Then, when DVC tries to access your S3 storage via `boto`, you should be able to
+get past SSL verification!
+
+### [Q: I want to be able to make my own plots in Python with data points from my `dvc plots`, including older versions of those plots. What do you recommend to get the raw historical data?](https://discord.com/channels/485586884165107732/563406153334128681/799617584336338954)
+
+We suggest
+
+```python
```

Comment:
could you please try `python3` here?

Reply:
done

Reply:
it didn't fix anything, still no smart highlighting

Reply:
with ```python3, no highlighting. so has to be ```python.